### PR TITLE
[NodeKiller] Change rounding of a number of selected nodes

### PR DESCRIPTION
**Pull request by jprzychodzen**
Ref. #1005

### CODE REVIEW

File: clusterloader2/pkg/chaos/nodes.go [hunk 1]
```diff
@@ -98,11 +99,15 @@ func (k *NodeKiller) pickNodes() ([]v1.Node, error) {
...
 		nodes[i], nodes[j] = nodes[j], nodes[i]
 	})
-	numNodes := int(k.config.FailureRate * float64(len(nodes)))
-	if len(nodes) > numNodes {
-		return nodes[:numNodes], nil
+	expectedNodesToFail := k.config.FailureRate * float64(len(nodes))
+	nodesToFail := nodes[:0]
+	klog.Infof("%s: %d nodes available, expecting to fail %f nodes", k, len(nodes), expectedNodesToFail)
+	for _, node := range nodes {
+		if rand.Float64() < k.config.FailureRate {
```
Comment (mm4tt) [inline clusterloader2/pkg/chaos/nodes.go]: This is already randomized, we call rand.Shuffle in line 99

File: clusterloader2/pkg/chaos/nodes.go [hunk 2]
```diff
@@ -124,7 +129,7 @@ func (k *NodeKiller) kill(nodes []v1.Node) {
 			time.Sleep(time.Duration(k.config.SimulatedDowntime))
 
 			klog.Infof("%s: Rebooting %q to repair the node", k, node.Name)
-			err = util.SSH("sudo reboot", &node, nil)
+			err = util.SSH("nohup sudo reboot +1s > /dev/null 2> /dev/null < /dev/null &", &node, nil)
```
Comment (mm4tt) [inline clusterloader2/pkg/chaos/nodes.go]: It could be a different PR (or at least commit).
Also, could you explain how it works? Ideally, in a comment. Personally, I don't understand the +1s part

File: clusterloader2/pkg/chaos/nodes.go [hunk 3]
```diff
@@ -98,11 +99,15 @@ func (k *NodeKiller) pickNodes() ([]v1.Node, error) {
 	rand.Shuffle(len(nodes), func(i, j int) {
 		nodes[i], nodes[j] = nodes[j], nodes[i]
 	})
-	numNodes := int(k.config.FailureRate * float64(len(nodes)))
-	if len(nodes) > numNodes {
-		return nodes[:numNodes], nil
+	expectedNodesToFail := k.config.FailureRate * float64(len(nodes))
+	nodesToFail := nodes[:0]
+	klog.Infof("%s: %d nodes available, expecting to fail %f nodes", k, len(nodes), expectedNodesToFail)
```
Comment (mm4tt) [inline clusterloader2/pkg/chaos/nodes.go]: +1 for adding more logs (I'd even add more, e.g. listing exactly which nodes were returned here)

File: clusterloader2/pkg/chaos/nodes.go (hunk 2 above)
Comment (jprzychodzen) [reply clusterloader2/pkg/chaos/nodes.go]: Sure, moved to  PR #1109  with extensive comment.

File: clusterloader2/pkg/chaos/nodes.go (hunk 1 above)
Comment (jprzychodzen) [reply clusterloader2/pkg/chaos/nodes.go]: They are randomized, but the size is constant.

And if we are expecting that less than a single node will fail (as for load test, with failure rate 0.01, and less than 100 eligible nodes, as some of them are running Prometheus), no node is scheduled.

File: clusterloader2/pkg/chaos/nodes.go (hunk 3 above)
Comment (jprzychodzen) [reply clusterloader2/pkg/chaos/nodes.go]: This information is available from 'kill' function in line 117, but I can duplicate that.

File: clusterloader2/pkg/chaos/nodes.go (hunk 1 above)
Comment (mm4tt) [reply clusterloader2/pkg/chaos/nodes.go]: I see, good debugging!
But, wouldn't be better to simply set numNodes to 1 if it's 0?
What I don't like about your solution is that, while it'll kill the `FailureRate` of nodes on average, we'll have test-runs where no nodes were killed and we'll have runs where 1+ nodes were killed. It's quite likely that it will make the test flaky - most likely it won't be failing but it can result in big variance of many metrics.

**Comment by jprzychodzen** [comment]
[Persubmit job does not have chaos monkey override enabled](https://github.com/kubernetes/test-infra/blob/308dab8f7f91eb1afccb5f174ca7c1f73f4d28b7/config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml#L305) , so this code is never invoked at this job.

File: clusterloader2/pkg/chaos/nodes.go (hunk 1 above)
Comment (jprzychodzen) [reply clusterloader2/pkg/chaos/nodes.go]: Sure, changed that to get more stable tests. Also, increased minimal interval in load test to prevent overkilling.

**Comment by jprzychodzen** [comment]
Fixes #1005
